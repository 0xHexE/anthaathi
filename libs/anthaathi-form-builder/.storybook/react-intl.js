const locales = ['en-US'];

const messages = locales.reduce(
  (acc, lang) => ({
    ...acc,
    [lang]: import(`../locales/compiled-locales/${lang}.json`), // whatever the relative path to your messages json is
  }),
  {},
);

const formats = {};

export const reactIntl = {
  defaultLocale: 'en-US',
  locales,
  messages,
  formats,
};
