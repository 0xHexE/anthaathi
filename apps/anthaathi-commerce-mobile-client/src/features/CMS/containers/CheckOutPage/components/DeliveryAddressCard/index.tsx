import React from 'react';
import {useIntl} from 'react-intl';
import {Pressable, View} from 'react-native';
import {Card, Text} from 'react-native-paper';

export interface DeliveryAddressCardProps {
  deliveryTitle: string;
  deliveryAddress: string;
  buttonTitle: string;
  handlePress?: () => void;
  mobileNumber: string;
}

const DeliveryAddressCard = (props: DeliveryAddressCardProps) => {
  const intl = useIntl();
  return (
    <Card
      style={{
        marginHorizontal: 10,
        borderColor: '#E3E2E7',
        borderWidth: 1,
        borderRadius: 4,
        marginVertical: 5,
      }}>
      <Card.Content>
        <Text style={{color: '#364A15', fontSize: 16, fontWeight: '600'}}>
          {props.deliveryTitle}
        </Text>
        <View style={{flexDirection: 'row', marginVertical: 5}}>
          <Text style={{color: '#364A15', fontSize: 14, width: '85%'}}>
            {props.deliveryAddress}
          </Text>
          <Pressable
            style={{paddingHorizontal: 10}}
            onPress={props.handlePress}>
            <Text style={{color: '#008D3E', fontSize: 14, fontWeight: '500'}}>
              {props.buttonTitle}
            </Text>
          </Pressable>
        </View>
        <View style={{flexDirection: 'row', marginVertical: 5}}>
          <Text style={{color: '#808080', fontSize: 14}}>
            {intl.formatMessage({defaultMessage: 'Mobile'})}
          </Text>
          <Text style={{color: '#364A15', fontSize: 14}}>
            {': ' + props.mobileNumber}
          </Text>
        </View>
      </Card.Content>
    </Card>
  );
};

export default DeliveryAddressCard;
