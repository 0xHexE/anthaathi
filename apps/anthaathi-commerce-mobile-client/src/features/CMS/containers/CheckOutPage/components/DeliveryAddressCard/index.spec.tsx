import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import DeliveryAddressCard from './index';

describe('DeliveryAddressCard', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddressCard
            deliveryTitle="Delivery Address"
            deliveryAddress="USA"
            buttonTitle="Change"
            mobileNumber="+90909090"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('deliveryAddressCard')).toBeTruthy();
  });
});
