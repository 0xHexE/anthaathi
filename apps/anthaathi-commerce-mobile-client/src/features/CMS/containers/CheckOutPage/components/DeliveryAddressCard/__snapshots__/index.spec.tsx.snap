// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryAddressCard should render the item 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#ffffff",
      "borderColor": "#E3E2E7",
      "borderRadius": 4,
      "borderWidth": 1,
      "elevation": 1,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 0.5,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 0.75,
    }
  }
>
  <View
    accessibilityState={
      {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="deliveryAddressCard"
  >
    <View
      style={
        [
          {
            "paddingHorizontal": 16,
          },
          {
            "paddingVertical": 16,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "#364A15",
              "fontSize": 16,
              "fontWeight": "600",
            },
          ]
        }
      >
        Delivery Address
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "color": "#364A15",
                "fontSize": 14,
                "width": "85%",
              },
            ]
          }
        >
          USA
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "paddingHorizontal": 10,
            }
          }
          testID="changeAddressButton"
        >
          <Text
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "#008D3E",
                  "fontSize": 14,
                  "fontWeight": "500",
                },
              ]
            }
            testID="deliverAddressButtonTitle"
          >
            Change
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "color": "#808080",
                "fontSize": 14,
              },
            ]
          }
        >
          Mobile
        </Text>
        <Text
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              {
                "color": "#364A15",
                "fontSize": 14,
              },
            ]
          }
        >
          : +90909090
        </Text>
      </View>
    </View>
  </View>
</View>
`;
